BrightFutures/Future.swift

```diff
@@ -383,12 +379,7 @@ public extension Future {
     /// Returns self
     public func onSuccess(context c: ExecutionContext = executionContextForCurrentContext(), callback: SuccessCallback) -> Future<T> {
         self.onComplete(context: c) { result in
-            switch result {
-            case .Success(let val):
-                callback(val.value)
-            default:
-                break
-            }
+            result.analysis(ifSuccess: callback, ifFailure: { _ in })
```

Comment:
Trivia Time™: the motivation for `analysis` is twofold:

1. It mirrors Haskell’s `either` function.
2. It allows you to take advantage of type inferencing in cases where `switch` statements do not.